### ISSUE TITLE
Feature idea: Latency simulation

### ISSUE THREAD

**Issue by nickluger** [react-cosmos/react-cosmos#654]
### Feature idea: Latency for Apollo or XHR

Simulate latency for responses, e.g. `resolveWith` in `react-cosmos-apollo-proxy`. Useful for:

- show loading indicators (e.g spinner)
- check for async issues

###  Example

Before we switched to `react-cosmos`, we used https://github.com/siteone/apollo-mocknetworkinterface before, which has the ability to simulate a timeout for example:
```js
export const ApolloMockProvider = ({ onRequest, latency, children }) => {
  const mockedNetworkFetch = createMockedNetworkFetch(onRequest, { latency });
  const client = new ApolloClient({
    link: createHttpLink({ uri: process.env.GRAPHQL_ENDPOINT, fetch: mockedNetworkFetch }),
    cache: new InMemoryCache(),
  });
  return (
    <ApolloProvider client={client}>;
      {children}
    </ApolloProvider>
  );
};
```

**`onRequest` is basically the same resolver-function-mock as `resolveWith` in cosmos.**

**Comment by dpehrson** [comment]
Is this required just to simulate loading? I finally got a chance to wire up the apollo proxy and it's amazing how well it works. The only issue we found was that it doesn't appear that you can simulate (and pin) the loading state of the data.

From what I see it renders once with `data.loading` set to true from the fixture, but almost immediately re-renders with it set to false.

I wanted to open an issue about that, but figured I should ask here first if this feature request is already being tracked.

**Comment by skidding** [comment]
> I wanted to open an issue about that, but figured I should ask here first if this feature request is already being tracked.

Not being tracked elsewhere as far as I know, so we can keep this thread open around this topic.

I don't have much insight into Apollo, so it's not clear to me how easy it would be to integrate latency into react-cosmos-apollo-proxy. @xavczen if you have any thoughts let us know.

**Comment by xavczen** [comment]
That seems doable! I imagine we could wrap [this block (code)](https://github.com/react-cosmos/react-cosmos/blob/master/packages/react-cosmos-apollo-proxy/src/fixtureLink.js#L39-L58) with a `setTimeout` whose value would be configured in the fixture!

I'm going to look into it this week 👍

**Comment by dpehrson** [comment]
@xavczen Excellent, would it be possible to set the timeout to something like `-1` which would indicate "never resolve"? The hope is for us to create a fixture that explicitly shows the loading state indefinitely so QA can do design acceptance on that state.

**Comment by nickluger** [comment]
If the fake resolver would be awaited as an `async` function, no separate latency-API would be necessary, as one could just timeout it. 

```js
const delay = ms => new Promise(resolve => setTimeout(resolve, ms))
// ...
 {
      resolveWith: async ({ variables: { input } }) => {
         await delay(5000);
     // etc.
```

Is that possible?

**Comment by xavczen** [comment]
@nickluger Yep that's possible, `resolveWith` can be resolved as a promise

**Comment by skidding** [comment]
Released @xavczen's PR under `4.7.0-3`. Try it out!
